io/kubernetes/pkg/cloudprovider/providers/gce EnsureInternalBackendServiceGroups/GetRegionBackendService_failed is flaky

Seeing several of the tests in ./pkg/cloudprovider/providers/gce flake 1 in 5-10 runs when run locally (I am logged in to gcloud).  Is it possible one of the APIs is not correctly shimmed and is fetching real data?

```
go test ./pkg/cloudprovider/providers/gce/ -test.run=TestEnsureInternalBackendServiceGroups/GetRegionBackendService_failed -count=10
--- FAIL: TestEnsureInternalBackendServiceGroups (0.00s)
    --- FAIL: TestEnsureInternalBackendServiceGroups/GetRegionBackendService_failed (0.00s)
    	gce_loadbalancer_internal_test.go:139:
    			Error Trace:	gce_loadbalancer_internal_test.go:139
    			Error:      	Not equal:
    			            	expected: []*compute.Backend{(*compute.Backend)(0xc420316e10), (*compute.Backend)(0xc420316ea0)}
    			            	actual  : []*compute.Backend{(*compute.Backend)(0xc420317290)}

    			            	Diff:
    			            	--- Expected
    			            	+++ Actual
    			            	@@ -1,2 +1,2 @@
    			            	-([]*compute.Backend) (len=2) {
    			            	+([]*compute.Backend) (len=1) {
    			            	  (*compute.Backend)({
    			            	@@ -5,16 +5,3 @@
    			            	   Description: (string) "",
    			            	-  Group: (string) (len=15) "new-test-node-1",
    			            	-  MaxConnections: (int64) 0,
    			            	-  MaxConnectionsPerInstance: (int64) 0,
    			            	-  MaxRate: (int64) 0,
    			            	-  MaxRatePerInstance: (float64) 0,
    			            	-  MaxUtilization: (float64) 0,
    			            	-  ForceSendFields: ([]string) <nil>,
    			            	-  NullFields: ([]string) <nil>
    			            	- }),
    			            	- (*compute.Backend)({
    			            	-  BalancingMode: (string) "",
    			            	-  CapacityScaler: (float64) 0,
    			            	-  Description: (string) "",
    			            	-  Group: (string) (len=15) "new-test-node-2",
    			            	+  Group: (string) (len=118) "https://www.googleapis.com/compute/v1/projects/test-project/zones/us-central1-b/instanceGroups/k8s-ig--test-cluster-id",
    			            	   MaxConnections: (int64) 0,
    			Test:       	TestEnsureInternalBackendServiceGroups/GetRegionBackendService_failed
FAIL
FAIL	k8s.io/kubernetes/pkg/cloudprovider/providers/gce	0.042s
```

@kubernetes/sig-cloud-provider-bugs

--- a/pkg/cloudprovider/providers/gce/gce_loadbalancer_internal_test.go
+++ b/pkg/cloudprovider/providers/gce/gce_loadbalancer_internal_test.go
@@ -80,6 +80,8 @@ func TestEnsureInternalBackendServiceUpdates(t *testing.T) {
 }
 
 func TestEnsureInternalBackendServiceGroups(t *testing.T) {
+	t.Parallel()
+
 	for desc, tc := range map[string]struct {
 		mockModifier func(*cloud.MockGCE)
 	}{
@@ -96,8 +98,6 @@ func TestEnsureInternalBackendServiceGroups(t *testing.T) {
 		},
 	} {
 		t.Run(desc, func(t *testing.T) {
-			t.Parallel()
-
 			vals := DefaultTestClusterValues()
 			nodeNames := []string{"test-node-1"}
 

--- a/pkg/cloudprovider/providers/gce/gce_loadbalancer_internal_test.go
+++ b/pkg/cloudprovider/providers/gce/gce_loadbalancer_internal_test.go
@@ -117,12 +117,12 @@ func TestEnsureInternalBackendServiceGroups(t *testing.T) {
 			err = gce.ensureInternalBackendService(bsName, "description", svc.Spec.SessionAffinity, cloud.SchemeInternal, "TCP", igLinks, "")
 			require.NoError(t, err)
 
-			// Update the BackendService with new Instances
+			// Update the BackendService with new InstanceGroups
 			if tc.mockModifier != nil {
 				tc.mockModifier(gce.c.(*cloud.MockGCE))
 			}
-			newNodeNames := []string{"new-test-node-1", "new-test-node-2"}
-			err = gce.ensureInternalBackendServiceGroups(bsName, newNodeNames)
+			newIGLinks := []string{"new-test-ig-1", "new-test-ig-2"}
+			err = gce.ensureInternalBackendServiceGroups(bsName, newIGLinks)
 			if tc.mockModifier != nil {
 				assert.Error(t, err)
 				return
@@ -132,10 +132,8 @@ func TestEnsureInternalBackendServiceGroups(t *testing.T) {
 			bs, err := gce.GetRegionBackendService(bsName, gce.region)
 			assert.NoError(t, err)
 
-			// Check that the instances are updated
-			newNodes, err := createAndInsertNodes(gce, newNodeNames, vals.ZoneName)
-			newIgLinks, err := gce.ensureInternalInstanceGroups(igName, newNodes)
-			backends := backendsFromGroupLinks(newIgLinks)
+			// Check that the Backends reflect the new InstanceGroups
+			backends := backendsFromGroupLinks(newIGLinks)
 			assert.Equal(t, bs.Backends, backends)
 		})
 	}

